system_metrics_collector/src/moving_average_statistics/types.hpp

```diff
@@ -37,7 +37,7 @@ struct StatisticData
  * @return std::string formatted struct contents in an easily readable format, e.g.,
  * /"avg=1, min=2, max=3, std_dev=4, count=5/"
  */
-static std::string statisticsDataToString(StatisticData results)
+static std::string statisticsDataToString(const StatisticData & results)
```

Comment:
This copies the function into the local file scope, which then gets inlined.

I'm not 100% sure here, but I think its more proper to use the modifier `inline` since the intent is that the function is to be copied in.

If the intent is that the function should be non-exportable, it should instead be declared inside an anonymous namespace.

Reply:
I'll leave it alone for now as discussed in person.